rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2672,6 +2681,91 @@ void voteFederationChange_add100Members_returnsSuccessResponseCodeAndFedSize100(
+            federationSupport.voteFederationChange(tx2, createFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            byte[][] commitFederationEventTopic = BridgeEvents.COMMIT_FEDERATION.getEvent().encodeEventTopics();
+            List<DataWord> encodedTopics = LogInfo.byteArrayToList(commitFederationEventTopic);
+            DataWord expectedCommitTopic = encodedTopics.get(0);
+
+            int EXPECTED_COUNT_OF_MEMBERS = 10;
+
+            // Voting add new fed with m of n authorizers
+
+            Set<String> expectedPubKeys = new HashSet<>();
+
+            for(int i = 0; i < EXPECTED_COUNT_OF_MEMBERS; i++) {
+                BtcECKey expectedBtcECKey = new BtcECKey();
+                expectedPubKeys.add(HexUtils.toJsonHex(expectedBtcECKey.getPubKey()));
+                ABICallSpec addFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.ADD.getKey(), new byte[][]{expectedBtcECKey.getPubKey()});
+                federationSupport.voteFederationChange(tx, addFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+                federationSupport.voteFederationChange(tx2, addFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            }
+
+            Keccak256 pendingFederationHash = federationSupport.getPendingFederationHash();
+
+            ABICallSpec commitFederationAbiCallSpec = new ABICallSpec(FederationChangeFunction.COMMIT.getKey(), new byte[][]{pendingFederationHash.getBytes()});
+
+            Federation initialActiveFederation = federationSupport.getActiveFederation();
+
+            // Act
+
+            // Voting commit new fed with m of n authorizers
+            federationSupport.voteFederationChange(tx, commitFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+            federationSupport.voteFederationChange(tx2, commitFederationAbiCallSpec, signatureCache, bridgeEventLogger);
+
+            // Assert
+
+            assertNull(federationSupport.getPendingFederationHash());
+            // -1 because the pending fed no longer exist
+            assertThat(federationSupport.getPendingFederationSize(), is(-1));
+
+            Federation finalActiveFederation = federationSupport.getActiveFederation();
+
+            assertThat(federationSupport.getActiveFederationSize(), is(EXPECTED_COUNT_OF_MEMBERS));
+
+            assertNotEquals(initialActiveFederation, finalActiveFederation);
+
+            Set<String> actualPubKeys = finalActiveFederation.getMembers().stream()
+                .map(member -> HexUtils.toJsonHex(member.getBtcPublicKey().getPubKey()))
+                .collect(Collectors.toSet());
+
+            assertEquals(expectedPubKeys, actualPubKeys);
+
+            assertThat(logs.size(), is(1));
+
+            LogInfo log = logs.get(0);
+            DataWord actualTopic = log.getTopics().get(0);
+
+            assertEquals(expectedCommitTopic, actualTopic);
```

Comment:
maybe adding a comment above like _assert that logCommitFederation was emitted with expected values_, wdyt?

Reply:
Updated.